--- mod_mono-1.2.4/configure.in.orig	2007-04-20 06:15:26.000000000 +0200
+++ mod_mono-1.2.4/configure.in	2007-05-16 20:21:28.673112778 +0200
@@ -82,6 +82,28 @@
   fi
 ],)
 
+# check for --with-apu-config
+AC_ARG_WITH(apu-config,
+[  --with-apu-config=PATH  Path to apr-config (apache 2.0). You may use
+                          this option when apu-config is not in the same
+			  directory as apxs and the output of
+			  'apu-config --includes' is different from
+			  'apxs -q INCLUDEDIR'.
+],
+[
+  if test -x "$withval"
+  then
+    AC_MSG_RESULT([$withval executable, good])
+    APUCONFIG=$withval
+  else
+    if test "xno" = "x$withval" ; then
+    	AC_MSG_RESULT([no])
+    else
+        AC_MSG_ERROR([$withval not found or not executable])
+    fi
+  fi
+],)
+
 if test -z "$APXS"; then
   for mydir in /usr/local/apache2/sbin	\
   		/usr/local/apache2/bin	\
@@ -172,6 +194,7 @@
 # apu-config - libapr-util can have different prefixes than libapr, check from apu-1-config to make sure
 #  (suse 10.1 is an example of this)
 
+if test -z "$APUCONFIG"; then
 for mydir in `dirname ${APXS}` \
 		/usr/local/apache2/sbin	\
   		/usr/local/apache2/bin	\
@@ -200,6 +223,7 @@
 		AC_MSG_RESULT(no)
 	fi
 done
+fi
 
 # last resort 1
 if test -z "$APUCONFIG"; then
