--- mod_mono-3.12/configure.in.orig	2015-01-12 18:50:02.000000000 +0100
+++ mod_mono-3.12/configure.in	2015-01-26 21:20:57.164211692 +0100
@@ -1,6 +1,6 @@
 AC_INIT(src/mod_mono.c)
 AM_INIT_AUTOMAKE(mod_mono, 3.12)
-AM_CONFIG_HEADER(include/mod_mono_config.h:config.in)
+AC_CONFIG_HEADERS(include/mod_mono_config.h:config.in)
 
 AC_PROG_CC
 AM_PROG_LIBTOOL
@@ -15,6 +15,7 @@
         AC_MSG_ERROR([You need to install pkg-config])
 fi
 
+m4_pattern_allow([^PKG_PATH$])
 PKG_PATH=""
 AC_ARG_WITH(crosspkgdir, [  --with-crosspkgdir=/path/to/pkg-config/dir],
         if test x$with_crosspkgdir = "x"; then
